### PR TITLE
Add no redirects

### PR DESCRIPTION
**Pull request by AsafMah**
### Security
* No redirects are allowed by default.

### CODE REVIEW

File: kusto/test/etoe/etoe_test.go [hunk 1]
```diff
@@ -1967,6 +1971,52 @@ func assertErrorsMatch(t *testing.T, got, want error) bool {
...
+
+			// Change field using reflection.
+			endQuery := reflect.
+				ValueOf(client).
+				Elem().
+				FieldByName("conn").
+				Elem().
+				Elem().
+				FieldByName("endQuery")
+			reflect.NewAt(endQuery.Type(), goUnsafe.Pointer(endQuery.UnsafeAddr())).Elem().Set(reflect.ValueOf(urll))
```
Comment (yogilad) [inline kusto/test/etoe/etoe_test.go:2009]: Why do you need this level of complexity?
Can't you return an KustoClientError with an explicit message saying Redirects Not Allowed?
Mind this is a requirement, or users will fail without knowing why.

File: kusto/cloudinfo.go [hunk 2]
```diff
@@ -64,8 +65,10 @@ func GetMetadata(kustoUri string, httpClient *http.Client) (CloudInfo, error) {
 		if err != nil {
 			return CloudInfo{}, err
 		}
-
-		u.Path = metadataPath
+		if !strings.HasPrefix(u.Path, "/") {
+			u.Path = "/" + u.Path
+		}
+		u = u.JoinPath(metadataPath)
```
Comment (yogilad) [inline kusto/cloudinfo.go:71]: Do you need to prepand '/' when using url.JoinPath?

**Review by yogilad — APPROVED** [review]

File: kusto/cloudinfo.go (hunk 2 above)
Comment (AsafMah) [reply kusto/cloudinfo.go:71]: For some reason, the "JoinPath" gives unexpected results without it.
Maybe I should add a comment?

Comment (yogilad) [reply kusto/cloudinfo.go:71]: won't hurt

**Pull request merged**